draft-ietf-quic-transport.md

```diff
@@ -1451,33 +1454,279 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+PATH_CHALLENGE frames or after some time has passed.  When setting this timer,
+implementations are cautioned that the new path could have a longer round-trip
+time than the original.
+
+Note that the endpoint might receive packets containing other frames on the new
+path, but a PATH_RESPONSE frame with appropriate data is required for path
+validation to succeed.
+
+If path validation fails, the path is deemed unusable.  This does not
+necessarily imply a failure of the connection - endpoints can continue sending
+packets over other paths as appropriate.  If no paths are available, an endpoint
+can wait for a new path to become available or close the connection.
+
+A path validation might be abandoned for other reasons besides
+failure. Primarily, this happens if a connection migration to a new path is
+initiated while a path validation on the old path is in progress.
+
+
 ## Connection Migration {#migration}
 
-A QUIC connection can be identified using the Destination Connection ID included
-in packets.  QUIC's consistent connection ID allows connections to survive
-changes to the client's IP and/or port, such as those caused by client or server
-migrating to a new network.  Connection migration allows a client to retain any
-shared state with a connection when they move networks.  This includes state
-that can be hard to recover such as outstanding requests, which might otherwise
-be lost with no easy way to retry them.
+QUIC allows connections to survive changes to endpoint addresses (that is, IP
+address and/or port), such as those caused by a endpoint migrating to a new
+network.  This section describes the process by which an endpoint migrates to a
+new address.
+
+An endpoint MUST NOT initiate connection migration before the handshake is
+finished and the endpoint has 1-RTT keys.
+
+This document limits migration of connections to new client addresses.
+Migrating a connection to a new server address is left for future work. If a
+client receives packets from a new server address, the client MAY discard these
+packets.
+
+
+### Probing a New Path
+
+An endpoint MAY probe for peer reachability from a new local address using path
+validation {{migrate-validate}} prior to migrating the connection to the new
+local address.  Failure of path validation simply means that the new path is not
+usable for this connection.  Failure to validate a path does not cause the
+connection to end unless there are no valid alternative paths available.
+
+An endpoint uses a new connection ID for probes sent from a new local address,
+see {{migration-linkability}} for further discussion.
+
+Receiving a PATH_CHALLENGE frame from a peer indicates that the peer is probing
+for reachability on a path. An endpoint sends a PATH_RESPONSE in response as per
+{{migrate-validate}}.
+
+PATH_CHALLENGE, PATH_RESPONSE, and PADDING frames are "probing" frames. If an
+endpoint has received packets containing only these frames from a given peer
```

Comment:
"Until an endpoint receives packets containing other frames from a given peer address, it MUST NOT [...]"